src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -386,6 +376,26 @@ private function addServiceToAmbiguousType($id, $type)
         $this->ambiguousServiceTypes[$type][] = $id;
     }
 
+    /**
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return \ReflectionMethod[]
+     */
+    private static function getSetters(\ReflectionClass $reflectionClass)
+    {
+        $setters = array();
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
+            $name = $reflectionMethod->getName();
+            if (isset($methodsCalled[$name]) || $reflectionMethod->isStatic() || 1 !== $reflectionMethod->getNumberOfParameters() || 0 !== strpos($name, 'set')) {
```

Comment:
initialisation of `$methidsCalled` is missing, and you never set it once you get out of the if condition, so `isset($methodscalled[$name])` should always return false